designdocs/Deterministic wallets.txt

```diff
@@ -198,27 +202,6 @@ that encode the index of each child as the tree is traversed downwards, with the
 private derivation (see the BIP32 spec for more information on this).
 
 
-Upgrade
--------
-
-HD wallets are strictly superior to old random wallets, thus by default all new wallets will be HD. The deterministic
-key chain will be created on demand by the KeyChainGroup, which allows the default parameters like lookahead size to
-be configured after construction of the wallet but before the DeterministicKeyChain is constructed.
-
-For old wallets that contain random keys, attempts to use any methods that rely on an HD chain being present will
-either automatically upgrade the wallet to HD, or if encrypted, throw an unchecked exception until the API user invokes
-an upgrade method that takes the users encryption key. The upgrade will select the oldest non-rotating private key,
-truncate it to 128 bits and use that as the seed for the new HD chain. We truncate and thus lose entropy because
-128 bits is more than enough, and people like to write down their seeds on paper. 128 bit seeds using the BIP 39
-mnemonic code specification yields 12 words, which is a convenient size.
-
-As part of migrating to deterministic wallets, if the wallet is encrypted the wallet author is expected to test
-after load whether the wallet needs an upgrade, and call the upgrade method explicitly with the password.
-Note that attempting to create a spend will fail if the wallet is not upgraded, because it will attempt to retrieve a
-change key which is done deterministically in the new version of the code: thus an non-upgraded wallet is not very
-useful for more than viewing (unless the API caller explicitly overrides the change address behaviour using the
-relevant field in SendRequest of course).
-
```

Comment:
Why are you removing this? Are we going to just write entirely new text to replace it?

Reply:
Are you referring to the entire "Upgrade" section?

It mainly describes the upgrade from random keys to HD, and this upgrade doesn't exist any more. But I admit I was lazy, I should have documented the P2PKH to P2WPKH upgrade instead, which is very similar. Is that what you were thinking about?

Reply:
Yes.